Implement 3rd-party bundle-list

castulo opened the pull request:
Implementation of the `swupd 3rd-party bundle-list` command.

Signed-off-by: Castulo Martinez <castulo.martinez@intel.com>

Closes #1179

otaviobp commented on src/bundle.c:
```diff
@@ -158,7 +158,8 @@ int required_by(struct list **reqd_by, const char *bundle_name, struct manifest
 					info("%s", msg);
 				}
-				info(" - %s\n", name);
+				info(" - ");
```
Separate bug fixes or other improvements from main commit. It's hard to understand why this change is made here

castulo replied:
ok, will do

otaviobp commented on src/bundle_list.c:
```diff
@@ -151,21 +147,25 @@ static enum swupd_code list_local_bundles()
 		return SWUPD_COULDNT_LIST_DIR;
 	}
 
+	info("Installed bundles:\n");
```
You are changing one feature on list-bundles that is unrelated to 3rd_party. Do that in a separate commit.

castulo replied:
👍

otaviobp commented on test/functional/3rd-party/3rd-party-bundle-list-basic.bats:
```diff
@@ -45,16 +78,35 @@ test_setup() {
 
-@test "TPR020: List installed 3rd-party bundles from a specific repo" {
 
+@test "TPR020: List installed 3rd-party bundles from a specific repo" {
```
Don't add a test in a PR if it's not being covered.

otaviobp commented on test/functional/3rd-party/3rd-party-bundle-list-deps.bats:
```diff
@@ -0,0 +1,83 @@
+
+load "../testlib"
+
+global_setup() {
```
Why are some tests added in the commit: "Add basic tests for 3rd-party bundle-list" and some tests are added in "Implementing 3rd-party bundle-list"?

otaviobp commented on src/3rd_party_bundle_list.c:
```diff
@@ -107,9 +107,41 @@ static bool parse_options(int argc, char **argv)
 
+static void print_repo_header(const char *repo_name)
+{
+	const int HEADER = 18;
```
where's this 18 came from? Can't it be replaced by a sizeof()?
magic numbers are usually hard to maintain.

castulo replied:
It is the length of " 3rd-Party Repo:  ", I'll modify this so we don't have magic numbers

castulo commented:
V2:
- Removed magic number from print_repo_header()
- split changes unrelated to 3rd_party bundle-list implementation to a separate commit
- removed tests related to code that is not yet being implemented from this PR
